### PR TITLE
Sponsored Listing Backend

### PR DESCRIPTION
**Pull request by bbryanb333**
Added fields for sponsored
functions to sponsor and unsponsored
modified search to show sponsored first

### CODE REVIEW

File: SH4RE/SH4RE/ListingViewModel.swift [hunk 1]
```diff
@@ -25,6 +25,7 @@ struct Listing : Identifiable{
     var category: String = ""
     var address:Dictionary<String,Double> = [:]
     var ownerName:String = ""
+    var sponsored:Int = 0
```
Comment (americobarros) [inline SH4RE/SH4RE/ListingViewModel.swift:28]: why 1/0 instead of true false?

File: SH4RE/SH4RE/ListingViewModel.swift [hunk 2]
```diff
@@ -697,3 +732,11 @@ func getStatus(requestId: String, listingId: String, completion: @escaping(Int)-
         }
     })
 }
+
+func sponsorListing(lid: String){
+    documentUpdate(collectionPath: "Listings", documentID: lid, data: ["sponsored":1])
+}
+
+func unsponsorListing(lid: String){
```
Comment (americobarros) [inline SH4RE/SH4RE/ListingViewModel.swift:740]: I will probably not have a use case for this but its good to have.

**Review by americobarros — CHANGES_REQUESTED** [review]
Would it be possible to change the sponsored to a true-false case instead? If not, you can merge!

File: SH4RE/SH4RE/ListingViewModel.swift (hunk 1 above)
Comment (bbryanb333) [reply SH4RE/SH4RE/ListingViewModel.swift:28]: It's easier to do sorting with a numerical value. Also I think if we improved upon this feature we would need some sort of hueristic to determine how often to show the sponsored listing. ie listings that pay more would have a higher value

**Review by americobarros — APPROVED** [review]
if not chaging the T/F og sponsored then I think this is good to go

**Pull request merged**